src/main/java/net/snowflake/client/jdbc/SnowflakeUtil.java

```diff
@@ -237,13 +241,23 @@ static ColumnTypeInfo getSnowflakeType(
             new ColumnTypeInfo(Types.INTEGER, defaultIfNull(extColTypeName, "INTEGER"), baseType);
         break;
       case FIXED:
-        columnTypeInfo =
-            new ColumnTypeInfo(fixedColType, defaultIfNull(extColTypeName, "NUMBER"), baseType);
+        if (isVectorType) {
+          columnTypeInfo =
+                  new ColumnTypeInfo(Types.INTEGER, defaultIfNull(extColTypeName, "INTEGER"), baseType);
```

Comment:
I'm not getting it, if we have a vector type, shouldn't we return something about vectors, not numbers? Or this is internal type in vector?

Reply:
this is about internal tyle in Vector. So isVectorType means that it is in context of a vector.